pupil-spa/src/app/services/answer/answer.service.ts

```diff
@@ -15,8 +18,24 @@ export class AnswerService {
       answers = [];
     }
 
+    // Check for duplicates - we can only store one answer for each question
+    const isAlreadyAnswered = answers.filter(
+      x => {
+        if (x.sequenceNumber === answer.sequenceNumber
```

Comment:
why not `x => x.sequenceNumber === answer.sequenceNumber && x.factor1 === answer.factor1 && x.factor2 === answer.factor2` ?